File: src/main/java/com/freya02/botcommands/api/commands/prefixed/annotations/TextOption.java
package com.freya02.botcommands.api.commands.prefixed.annotations;

import com.freya02.botcommands.api.commands.annotations.Optional;
import org.jetbrains.annotations.Nullable;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation used to specify a text command parameter is supplied from a Discord message.
 * <br>This also can set name and example of {@linkplain JDATextCommand text commands} parameters
 * <p>
 * {@link #name()} is optional if the parameter name is available (add -parameters to your java compiler)
 *
 * @see Optional Optional (can also see @Nullable)
 * @see Nullable
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface TextOption {
	String name() default "";

	String example() default "";
}

File: src/main/kotlin/io/github/freya022/botcommands/api/commands/text/annotations/TextOption.kt
package io.github.freya022.botcommands.api.commands.text.annotations

import io.github.freya022.botcommands.api.commands.annotations.Optional
import org.jetbrains.annotations.Nullable

/**
 * Sets a parameter as a text command option from the Discord message.
 *
 * This also can set name and example of [text commands][JDATextCommandVariation] parameters.
 *
 * @see Optional @Optional
 * @see Nullable @Nullable
 * @see ID @ID
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class TextOption(
    /**
     * The name of this option displayed on the help content.
     *
     * This is optional if the parameter name is available,
     * see [the wiki](https://freya022.github.io/BotCommands/3.X/using-botcommands/parameter-names/) for more details.
     */
    val name: String = "",

    /**
     * The example input of this option displayed on the help content.
     */
    val example: String = ""
)
